### PR TITLE
Add travis support

### PR DESCRIPTION
**Pull request by ernestas-poskus**
@quux00 

Adding https://travis-ci.org/ support

This downloads OrientDB 2.0.13, installs it and runs or prepares for test

P.S. adopted from https://github.com/doctrine/orientdb-odm

with next PR will add *health* test

### CODE REVIEW

**Comment by quux00** [comment]
Cool - thanks so much.  However, I haven't taken the time to learn travis system yet.  If the tests start failing, how does one debug the issue?  Can one run them locally and reproduce the problem?

Also have you tested this set up?  I need to understand how to validate/test this before merging it. 

Thanks again!

**Comment by ernestas-poskus** [comment]
Travis is continuous integration system (like Jenkins if you are familiar with it).

It runs tests.

> Can one run them locally and reproduce the problem? In other words, if I run my own tests locally and they pass but the tests are failing in travis, how does one proceed to diagnose the problem and make a fix?

See the output below
https://travis-ci.org/linkosmos/meta

screenshot 2015-08-06 18 04 04 also there is a badge that indicates whether tests pass.

It is very useful for open-source projects because contributors can get instant feedback.

**Comment by ernestas-poskus** [comment]
> Also have you tested this set up? I need to understand how to validate/test this before merging it.

I have refactored your client.go file a bit and added new one client_test.go in order to run tests on travis as well as in local environment.

File: client.go [hunk 1]
```diff
@@ -54,12 +54,11 @@ import (
 //   to test the data statements only
 //
 
-// EDIT THESE to match your setup
-const (
```
Comment (ernestas-poskus) [inline client.go]: refactored to flags to avoid the necessity to change const's

**Comment by ernestas-poskus** [comment]
@quux00 

Please authorise travis to access this repository here is the guide http://docs.travis-ci.com/user/getting-started/

P.S. .travis.yml is already present

**Comment by ernestas-poskus** [comment]
you can inspect current build 

https://travis-ci.org/ernestas-poskus/ogonori/jobs/74429246

16.96s$ go test -cpu=2 -race -v ./...
=== RUN TestAgainstOrientDBServer-2
FATAL: client_test.go:36: EOF

it have failed due to data races which is normal, we will fix it :bug:

**Comment by quux00** [comment]
Thanks for the work here.  I will look at this closely tonight and give my feedback.

**Comment by quux00** [comment]
Hi ernestas-poskus,

Nice work here.  I had some pending new code that I wanted to get pushed out around UPDATES. I also made a bunch of changes based on `golint` and `govet` and pushed those up to GitHub.  This created merge conflicts with your pull request here.  So I pulled your changes down to my system and merged them in.

I also added

	// do not run functional tests if -short flag is used
	if testing.Short() {
		return
	}

to client_test.go.  This functional test can be very slow on machines with slow disk (such as my laptop).  This allows people to bypass the functional test by doing:

    go test -short ./...

I will be out of town most of this weekend, so feel free to make progress.

I have authorized travis to access this repo.  Thanks for getting me hooked up with this.  Been on my TODO list for a while.

**Comment by quux00** [comment]
I see that the travis builds are failing: https://travis-ci.org/quux00/ogonori/jobs/74660652

When I run `go test -cpu=2 -race -v ./...` locally it passes.  I don't have time to research it right now, but the client.go test (and presumably client_test.go) depends on having the data in `scripts/ogonori-setup.sql`
already in the db (unless you do `go run client.go create`).  Can you look into addressing that in the client_test.go?

**Comment by ernestas-poskus** [comment]
>  I don't have time to research it right now, but the client.go test (and presumably client_test.go) depends on having the data in scripts/ogonori-setup.sql

will check it

**Pull request merged**